src/main/java/com/moodysalem/phantomjs/wrapper/PhantomJSSetup.java

```diff
@@ -0,0 +1,215 @@
+package com.moodysalem.phantomjs.wrapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import com.moodysalem.phantomjs.wrapper.beans.OperatingSystem;
+
+public final class PhantomJSSetup{
+
+	protected final static Logger logger = Logger.getLogger(PhantomJSSetup.class.getName());
+
+	// this will store a reference to the executable phantomjs binary after we unzip the resource
+	protected static File PHANTOM_JS_BINARY = null;
```

Comment:
Yeah that should work-not as worried about the class being final since it's less likely IMO that someone would accidentally subclass it and cause issues vs. accidentally assigning the variable a new value